### PR TITLE
TRAD-443 Browser support (attempt 2)

### PR DESCRIPTION
**Pull request by aspin**
Changes:
- `node-fetch` => `axios` (for isomorphic cross-platform support)
- build pipeline changed (`esbuild` + use bundled files w/differences for platforms)
- reformatted code
- generated proto extension .js => empty string

Going to include a follow-up PR after to update the tests a bit more.

### CODE REVIEW

**Comment by aspin** [comment]
Review suggestions: 
- Ignore `bxsolana/proto` changes.
- Read `provider/*.ts` constructor changes. Initialization logic has changed a bit (in conjunction with `utils/config.ts`). The is mostly formatting and reordering the methods to make a bit more sense (e.g. grouping streaming methods, etc.)
- Read `bxsolana/index.*` additions.
- Read `build.mjs`, `package.json`, `README.md`

**Comment by aspin** [comment]
Screen Shot 2023-01-13 at 1 12 37 PM

The generated `dist/` directory looks like this

File: bxsolana/provider/base.ts [hunk 1]
```diff
@@ -1,30 +1,52 @@
 import {
```
Comment (atulsriv) [inline bxsolana/provider/base.ts:1]: this file lgtm

File: bxsolana/utils/config.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import dotenv from "dotenv"
```
Comment (atulsriv) [inline bxsolana/utils/config.ts:1]: this file lgtm

File: bxsolana/provider/grpc.ts [hunk 3]
```diff
@@ -34,67 +59,83 @@ import {
...
 export class GrpcProvider extends BaseProvider {
     private client: Service
     private grpcClient: Client
 
-    constructor(address = `${MAINNET_API_GRPC_HOST}:${MAINNET_API_GRPC_PORT}`, useTls: boolean) {
-        super()
-        const metaCallback = (options: CallMetadataOptions, cb: (err: Error | null, metadata?: grpc.Metadata) => void) => {
+    constructor(
+        authHeader: string,
+        privateKey = "",
```
Comment (atulsriv) [inline bxsolana/provider/grpc.ts:104]: define type as string?

File: bxsolana/provider/http.ts [hunk 4]
```diff
@@ -33,86 +34,128 @@ import {
...
+import axios, { AxiosResponse } from "axios"
 
 export class HttpProvider extends BaseProvider {
     private baseUrl: string
 
-    constructor(address: string = MAINNET_API_HTTP) {
-        super()
+    constructor(
+        authHeader: string,
+        privateKey = "",
```
Comment (atulsriv) [inline bxsolana/provider/http.ts:54]: same, define type as string

**Review by atulsriv — APPROVED** [review]

**Review by DimaCantemir — APPROVED** [review]

File: bxsolana/provider/grpc.ts (hunk 3 above)
Comment (aspin) [reply bxsolana/provider/grpc.ts:104]: ESLint actually complains about doing that kind of thing:

```
ESLint: Type string trivially inferred from a string literal, remove type annotation.(@typescript-eslint/no-inferrable-types)
```

**Pull request merged**